dask/utils_test.py

```diff
@@ -142,3 +143,18 @@ def hlg_layer(hlg: HighLevelGraph, prefix: str) -> Layer:
 def hlg_layer_topological(hlg: HighLevelGraph, i: int) -> Layer:
     "Get the layer from a HighLevelGraph at position ``i``, topologically"
     return hlg.layers[hlg._toposort_layers()[i]]
+
+
+@contextlib.contextmanager
+def _check_warning(
+    condition: bool, category: type[Warning], message: str | None = None
+):
+    """Conditionally check is a warning is raised"""
+    if condition:
+        import pytest
+
+        with pytest.warns(category, match=message) as ctx:
+            yield ctx
+    else:
+        with contextlib.nullcontext() as ctx:
+            yield ctx
```

Comment:
Defining a private utility may be overkill, but I can see it being useful, particularly for handling deprecations. @jsignell what do you think? Happy to inline elsewhere (like was done in https://github.com/dask/dask/pull/8653) or keep and add expand the docstring + add tests

Reply:
Oh this seems very useful to me, but can we make the message mandatory, It can be really hard to remove these kinds of checks when later if it's unclear what they were for.

Reply:
Done 👍